src/main/java/io/github/thebusybiscuit/sensibletoolbox/api/items/BaseSTBMachine.java

```diff
@@ -1074,7 +1075,9 @@ public void onServerTick() {
     }
 
     private double transferCharge(Chargeable from, Chargeable to) {
-        if (to.getCharge() >= to.getMaxCharge() || from.getCharge() == 0) {
+        if (MiscUtil.truncateDecimal(from.getCharge(), 2)
+                >= MiscUtil.truncateDecimal(to.getMaxCharge(), 2)
+                || MiscUtil.truncateDecimal(from.getCharge(), 2) == 0) {
```

Comment:
You're doing `MiscUtil.truncateDecimal(from.getCharge(), 2)` twice here, you can save yourself one of those operations by creating a variable for it.